criu/net.c

```diff
@@ -3171,18 +3185,44 @@ static inline int nftables_network_unlock(void)
 	struct nft_ctx *nft;
 	char table[32];
 	char buf[128];
+	FILE *fp;
+	uint64_t locking_run_id;
+
+	/**
+	 * Network unlocking can happen during restore, but also
+	 * during checkpointing. It happens during checkpointing
+	 * if the process is kept on running or if there was some
+	 * error. In that case, checkpointing, dump_criu_run_id
+	 * is not set and we have to use criu_run_id.
+	 */
 
-	if (nftables_get_table(table, sizeof(table)))
+	if (dump_criu_run_id == 0)
```

Comment:
I think we need to introduce a boolean parameter here to determine if we are on restore/dump codepath as `dump_criu_run_id == 0` might be in two different cases: when we deal with an old image (without `dump_criu_run_id` field) or if we are on the restore codepath.